UnitTest/tests/test_sub_SIGNALduino_Get-definition.txt

```diff
@@ -181,6 +181,91 @@ defmod test_sub_SIGNALduino_Get UnitTest dummyDuino
 		$SD_Parse->unmock;
 	}; 
 
+	subtest 'Test device with dummy attrib 1' => sub {
+		plan tests => 3;
 
+		subtest 'Test Attributes (hardware undef)' => sub {
+			plan tests => 2;
+			is(AttrVal($target, "dummy", 0),1,"check attrib dummy is 1");
+			delete $attr{$target}{hardware} if (AttrVal($target, "hardware", undef));
+			is(AttrVal($target, "hardware", undef),undef,"check attrib hardware undef");
+		};
+
+		subtest 'Test allowed - get commands' => sub {
+			my $rawArg="?";
+			my $ret=SIGNALduino_Get($targetHash, $target, "?", $rawArg);
+
+			my @tests = split(" ", substr($ret,index($ret,"one of")+7));
+			my $tests = scalar(@tests);
+			plan tests => $tests;
+
+			like($ret,qr/^Unknown argument \?, .*availableFirmware:noArg.*/,"check cmd availableFirmware accepted");
+			like($ret,qr/^Unknown argument \?, .*raw$/,"check cmd raw accepted");
+		};
+
+		subtest 'Test not allowed - get commands' => sub {
+			my $rawArg="?";
+			my $ret=SIGNALduino_Get($targetHash, $target, "?", $rawArg);
```

Comment:
Ich habe das gefühl, dass Du SIGNALduino_Get mit einem "?" Argument zu viel aufrufst...

Reply:
Das wird aus dem Wirrwar sein mit C&P bzw. nein den Test doch so.
Dennoch musste ich soeben feststellen dies ist erlaubt und es geht auch im FHEM WEB mit einem ? ?.

Bereinigen?

Reply:
Ich würde es bereinigen, weil die Ausgabe mit choose one of ... muss nur auf ? erfolgen und keine "Varianten" davon :)